### ISSUE TITLE
Distinguish between download and auth proxies

### ISSUE THREAD

**Issue by Dreamer269** [devine-dl/devine#29]
**Is your feature request related to a problem? Please describe.**
Some services need a proxy to use them. In my opinion, a distinction should be made here between a download and an auth proxy. Funnily enough, there are services where you only need a proxy for login, stream requests and license requests. They don't care from which IP you download the tracks from the mpd. Through the distinction, you can use the full bandwidth when downloading without having to route the whole traffic through the download proxy. This saves time and expensive data traffic.

**Describe the solution you'd like**
So far, it only checks whether the current track needs a proxy, which is then used for the download. So we could add the two functions track.needs_auth_proxy and track.needs_download_proxy and check these conditions at login, download and before the license request.
That would be really great. This is what keeps me from migrating more services to devine.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by rlaphoenix** [comment]
What you ask for is already implemented but seemingly misunderstood. The `GEOFENCE` defines if a proxy to a certain region is required for the service in general. Whether that is just authentication or downloading or both. The `Track.needs_proxy` variable defines if the same proxy should be used when downloading or not. By default, it is set to False.

Therefore, for what you need just do a:

```py
tracks = DASH.from_url(...).to_tracks(...)
# ...
for track in tracks:
    track.needs_proxy = True
```

You could even filter it by specific track types using `isinstance(track, (Video, Audio))` for example.

**Comment by Dreamer269** [comment]
Oh okay, thanks for the explanation. I'll mirgrate some services and test it.
Maybe you could add some FAQ or a documentation in the future. I think that will help many people who want to use devine. Especially if the questions go into detail. I would be more than happy to contribute to it :)

**Comment by Dreamer269** [comment]
I just tested it and it works great :) The only thing I am still curious about is, if it's possible to have the GEOFENCE always activated.
Let's say I'm in the US and want to rip from Hulu. In this case the GEOFENCE wouldn't request a proxy because my US IP is in the GEOFENCE list . But thats a problem, because Hulu blocked my IP. Therefore I would prefer to have the GEOFENCE always activated so that devine requests a proxy. I already tried to leave it empty, but that didn't work. 

My workaround would be to set the GEOFENCE to a fictional country like "always", so that my IP never matches it because the country always doesn't exist xD. But is this way really good?
